src/provisioner/cluster.py

```diff
@@ -48,10 +51,15 @@ def __init__(self, node_list: list = []):
 
     def _validate(self, node: dict):
         """
-        validates a give node to habve required properties
+        validates a give node to have required properties
         Raises exception if there is any entry missing
         """
-        pass
+
+        required_keys_for_node = [
+            'name', 'id', 'components', 'storage_set', 'hostname']
+        for k in required_keys_for_node:
+            if node.get(k) is None:
+                raise VError(errno.EINVAL, f'{k} key is missing in node dict.')
```

Comment:
Error should be clear enough to point to specific node. 
```
node_name = node.get('name')
if node_name is None:
     raise CortxProvisionerError(errno.EINVAL, 'Missing name for the node entry')
required_keys_for_node = [ 'id', 'components', 'storage_set', 'hostname']
for k in required_keys_for_node:
    if node.get(k) is None:
       raise CortxProvisionerError(errno.EINVAL, f'{k} property {k} unspecified for node {node_name}')
```

Reply:
Done.